include/battle_main.h

```diff
@@ -23,6 +23,26 @@ struct MultiPartnerMenuPokemon
 #define BOUNCE_MON          0x0
 #define BOUNCE_HEALTHBOX    0x1
 
+#if MODERN
+#define U8_ENUM : u8
+#else
+#define U8_ENUM
+#endif
+
+enum FirstTurnEventStates U8_ENUM {
+    FIRST_TURN_EVENTS_START,
+    FIRST_TURN_EVENTS_OVERWORLD_WEATHER,
+    FIRST_TURN_EVENTS_TERRAIN,
+    FIRST_TURN_EVENTS_STARTING_STATUS,
+    FIRST_TURN_EVENTS_TOTEM_BOOST,
+    FIRST_TURN_EVENTS_NEUTRALIZING_GAS,
+    FIRST_TURN_EVENTS_SWITCH_IN_ABILITIES,
+    FIRST_TURN_EVENTS_OPPORTUNIST_1,
+    FIRST_TURN_EVENTS_ITEM_EFFECTS,
+    FIRST_TURN_EVENTS_OPPORTUNIST_2,
+    FIRST_TURN_EVENTS_END,
+};
```

Comment:
If we're hoping to have this still be in master before 1.9, can we perhaps not use this to-be-figured-out enum system but stick with good old defines for now? We can obviously refactor this later, but it feels a bit out of place in the current ecosystem

Reply:
turned it into a simple enum